openmdao/docs/_exts/embed_code.py

```diff
@@ -169,9 +176,23 @@ def run(self):
                 if 'plot' in layout:
                     code_to_run = code_to_run + ('\nmatplotlib.pyplot.savefig("%s")' % plot_file_abs)
 
-            skipped, failed, run_outputs = run_code(code_to_run, path, module=module, cls=class_,
-                                                    imports_not_required=imports_not_required,
-                                                    shows_plot=shows_plot)
+            if is_test and decorators is not None:
+                # check whether test is skipped due to decorator
+                test_suite = unittest.TestLoader().loadTestsFromName(path)
+                test_result = unittest.TextTestRunner(stream=StringIO()).run(test_suite)
```

Comment:
What is the performance penalty for running each test twice like this whenever we build the docs?

Reply:
there is a hit no doubt, but the test has to run in the test context for the decorator and again in the isolated context for the docs..   I don't see a way around it if we want to build docs with skipped tests as described in the issue...  open to suggestions

Reply:
this is only done for decorated tests of course

Reply:
It turns out that if a method is going to be skipped, the method instance will have two attributes called '__unittest_skip__' and '__unittest_skip_why__', so you can simply check for the existence of '__unittest_skip__' for example and know that it'll be skipped without actually running the test.  Our current 'get_source_code' function already looks up the method object although it currently doesn't return it, but we could easily return it and look for the `__unittest_skip__` attribute to see if that test is going to be skipped.

Reply:
ok will look at this